src/classes/i18n.js

```diff
@@ -199,15 +209,94 @@ export default class i18n {
     }
 
     get_lang_files (lang, cb) {
+        const MAX_RETRY = this.config.get('download_retry');
 
         // append a random string, to avoid getting a response from cache
         const random_str = '&rand=' + ~~(Math.random() * 1000);
         const url = this.translations_url.replace(':lang', lang) + random_str;
-        const write_stream = fs.createWriteStream(this.config.get('locale_dir') + lang + '.json');
 
-        request(url)
-            .pipe(write_stream)
-            .on('finish', cb);
+        const translation_file_path = this.config.get('locale_dir') + lang + '.json';
+
+        new Promise(async (resolve, reject) => {
+            if (
+                process.env.REFRESH_TRANSLATIONS
+                || !await this.version_match(translation_file_path)
+            ) {
+                // force re-download translations
+                await this.download_translations(url, translation_file_path);
+            }
+
+
+            let retry_count = 0;
+
+            async.doWhilst(
+                async () => {
+                    try {
+                        await this.check_file(translation_file_path);
```

Comment:
This should/can be done right after we download the file?

Reply:
If we move this after download, that would not consider the case when the translation has no issues i.e. right version, not corrupted, etc and would always have to download the translation (which is what we're trying to avoid in the first place)